mixer/adapter/signalfx/tracing.go

```diff
@@ -225,18 +226,20 @@ func convertInstance(istioSpan *tracespan.Instance) *trace.Span {
 		span.LocalEndpoint.ServiceName = &n
 	}
 
-	if ip, ok := istioSpan.SpanTags["source.ip"].(net.IP); ok {
-		ips := ip.String()
+	if ip, ok := istioSpan.SpanTags["source.ip"].([]byte); ok {
+		ips := net.IP(ip).String()
 		span.LocalEndpoint.Ipv4 = &ips
+		tags["source.ip"] = ips
 	}
 
 	if n, ok := istioSpan.SpanTags["destination.name"].(string); ok && n != "" {
 		span.RemoteEndpoint.ServiceName = &n
 	}
 
-	if ip, ok := istioSpan.SpanTags["destination.ip"].(net.IP); ok {
-		ips := ip.String()
+	if ip, ok := istioSpan.SpanTags["destination.ip"].([]byte); ok {
```

Comment:
Why is net.IP being replaced in this code with []byte? We want the adapters to receive the stronger type signatures.

Reply:
It's a problem with in-process adapter code generation. Dynamic values are turned into raw attribute values.